compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -287,8 +287,9 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                         llval
                     }
                 };
-                let addr = bx.pointercast(llslot, bx.type_ptr_to(bx.cast_backend_type(&cast_ty)));
-                bx.load(addr, self.fn_abi.ret.layout.align.abi)
+                let llty = bx.cast_backend_type(&cast_ty);
```

Comment:
This is not necessarily a llvm type . Perhaps just `ty`? (though we call vals `llval` still...)